zqd/api/api.go

```diff
@@ -72,7 +72,16 @@ type ScannerStats struct {
 	RecordsMatched int64 `json:"records_matched"`
 }
 
+type SpaceID string
+
+type SpaceDesc struct {
```

Comment:
I had similar questions as noah about SpaceDesc vs SpaceInfo. At the moment, I see Brim making a lot of requests to get the full space info, as they're trying to watch for updates to the span & size during ingest, so there's no expectation that SpaceInfo is "a lot" of info to gather & return. Once we build in support for spaces backed by non-local files, we'll have a better idea of what's cheap for a SpaceList call vs not, so maybe we can hold off on this split until then.

Reply:
ok

Reply:
I'll do this but I do want to say this- I don't think this is about making SpaceList a cheaper call. As config options grow (they most certainly will) we are going to be adding more and more information to space info. It feels like there are circumstances where this is TMI. For instance the response in a SpacePost as well as SpaceList- where I mostly just care about the ID of the space and the name.

(All smaller issue to consider) This also adds additional complexity to tests where the expected response now has to account for larger range of variables.